### PR TITLE
Optimize CSS

### LINKED ISSUE #5171: Optimize Next CSS
Configure NUXT VSF Module to include:

- criticalCSS inlined

https://web.dev/extract-critical-css/

- PurgeCSS
https://github.com/Developmint/nuxt-purgecss

Post before/after results!!

Under `@vue-storefront/nuxt` config this should be marked a `performance.criticalCSS` and `performance.purgeCSS` properties defaulting to `true`

### PR DESCRIPTION
**Pull request by filipsobol**
### Short Description and Why It's Useful
This PR optimizes styles in two ways:

1. A lot of styles were duplicated by importing `~@storefront-ui/vue/styles` in every component that used variables or mixins from StorefrontUI library. Now thanks to `@nuxtjs/style-resources` module, all variables and mixins from `_helpers.scss` from SFUI will be automagically available in components without the need to import anything.
2. From now on `nuxt-purgecss` library will remove unused styles (**disabled by default**).

**First step reduced overall size of the home page by 350KB and second step by another 50KB** (see screenshots below). This also reduced overall page load time.

Results will be slightly better with improvements described in this issue: https://github.com/DivanteLtd/storefront-ui/issues/1534

**IMPORTANT:**
1. Because PurgeCSS looks for whole class names in files, it may remove styles for dynamic classes. If there is a dynamic class, please use whole names instead of concatinating variables (eg. `isDev ? 'some-style-dev' : 'some-style-prod'` instead of `some-style-${ isDev ? 'dev' : 'prod' }`. If this can't be avoided you can use `whitelist` property in PurgeCSS.
2. Do not import styles from `@storefront-ui` libraries - all variables and mixins are available as long as there is a `lang="scss"` attribute in `<style>`.
3. Do not add files with styles to `@nuxtjs/style-resources` configuration, only those that define variables, mixins etc. Please see this link for more information: https://github.com/nuxt-community/style-resources-module#warning

Closes #5171.

### Screenshots of Visual Changes before/after (if There Are Any)
Before:
before

After:
after

### Which Environment This Relates To
- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [X] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [ ] No upgrade steps required (100% backward compatibility and no breaking changes)
- [X] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

### CODE REVIEW

**Comment by filipsobol** [comment]
I converted this PR to draft, because there are some visual bugs on deployed preview.

**Review by justyna-13 — APPROVED** [review]

**Review by AdamPawlinski — APPROVED** [review]
It works perfectly. Only one question, should we remove this imports also from Shopify integration and CheckoutCom or not?

**Comment by filipsobol** [comment]
> It works perfectly. Only one question, should we remove this imports also from Shopify integration and CheckoutCom or not?

I don't know about `checkout-com`, but for Shopify we can't for now, because it currently uses old versions of `@vue-storefront/nuxt` and `@vue-storefront/nuxt-theme`

File: packages/core/docs/general/performance.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Performance
```
Comment (filrak) [inline packages/core/docs/general/performance.md:1]: in the (near) future we should have a big docs about this module that perf doc will be a part of :)

File: packages/core/nuxt-module/lib/modules/performance.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          .filter(({ asType }) => asType === 'script')
+          .map(({ file, asType }) => `<${publicPath}${file}>; rel=preload; as=${asType}`);
+      }
+    }
+  });
+}
+
+function loadPurgeCss(options) {
+  // PurgeCSS module should be installed after all other modules
+  this.nuxt.hook('modules:done', moduleContainer => moduleContainer.addModule([ 'nuxt-purgecss', options ]));
```
Comment (filrak) [inline packages/core/nuxt-module/lib/modules/performance.js:18]: if we are adding other modules we should definitely prompt about this

File: packages/core/docs/general/performance.md (hunk 1 above)
Comment (filipsobol) [reply packages/core/docs/general/performance.md:1]: That's a good idea 👍  For now I just wrote simple page just to have new functionalities documented at all

File: packages/core/nuxt-module/lib/modules/raw-sources-loader.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const log = require('../helpers/log');
+const resolveDependency = require('../helpers/resolveDependency');
+
+module.exports = function VueStorefrontPerformanceModule (options) {
+  const isProd = process.env.NODE_ENV === 'production' || options.coreDevelopment;
```
Comment (filrak) [inline packages/core/nuxt-module/lib/modules/raw-sources-loader.js:5]: we can lift this up to some helpers file

**Review by filrak — APPROVED** [review]

File: packages/core/nuxt-module/lib/modules/performance.js (hunk 2 above)
Comment (filipsobol) [reply packages/core/nuxt-module/lib/modules/performance.js:18]: `nuxt-purgecss` module already logs information to the console by itself.

**Pull request merged**